From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Microsoft Azure IoT <azure-iot@microsoft.com>
Date: Mon, 15 Dec 2024 00:00:00 +0000
Subject: [PATCH] Add ext2fs_file_get_current_physblock function

This function exposes the current physical block number from an open
e2fsprogs file handle. It is needed for efficient delta processing of
ext4 filesystems.

Upstream-Status: Inappropriate [Azure-specific]
Signed-off-by: Microsoft Azure IoT <azure-iot@microsoft.com>
---
 lib/ext2fs/ext2fs.h | 1 +
 lib/ext2fs/fileio.c | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/lib/ext2fs/ext2fs.h b/lib/ext2fs/ext2fs.h
index 6e87829f..a7935a5f 100644
--- a/lib/ext2fs/ext2fs.h
+++ b/lib/ext2fs/ext2fs.h
@@ -1474,6 +1474,7 @@ errcode_t ext2fs_file_get_lsize(ext2_file_t file, __u64 *ret_size);
 extern ext2_off_t ext2fs_file_get_size(ext2_file_t file);
 extern errcode_t ext2fs_file_set_size(ext2_file_t file, ext2_off_t size);
 extern errcode_t ext2fs_file_set_size2(ext2_file_t file, ext2_off64_t size);
+extern blk64_t ext2fs_file_get_current_physblock(ext2_file_t file);
 
 /* finddev.c */
 extern char *ext2fs_find_block_device(dev_t device);
diff --git a/lib/ext2fs/fileio.c b/lib/ext2fs/fileio.c
index 818f7f05..8e163cd2 100644
--- a/lib/ext2fs/fileio.c
+++ b/lib/ext2fs/fileio.c
@@ -664,3 +664,8 @@ errcode_t ext2fs_file_set_size(ext2_file_t file, ext2_off_t size)
 {
 	return ext2fs_file_set_size2(file, size);
 }
+
+blk64_t ext2fs_file_get_current_physblock(ext2_file_t file)
+{
+    return file->physblock;
+}
\ No newline at end of file
